esphome/components/atm90e32/atm90e32.cpp

```diff
@@ -396,6 +374,19 @@ void ATM90E32Component::dump_config() {
   LOG_SENSOR("  ", "Peak Current C", this->phase_[PHASEC].peak_current_sensor_);
   LOG_SENSOR("  ", "Frequency", this->freq_sensor_);
   LOG_SENSOR("  ", "Chip Temp", this->chip_temperature_sensor_);
+#ifdef USE_API
+  if (this->restored_offset_calibration_ || this->restored_power_offset_calibration_ ||
+      this->restored_gain_calibration_ || !this->enable_offset_calibration_ || !this->enable_gain_calibration_) {
+    this->set_interval("atm90e32_calibration_log", 1000, [this]() {
+#ifdef USE_NETWORK
+      if (!network::is_connected())
+        return;
+#endif
+      this->log_calibration_status_();
+      this->cancel_interval("atm90e32_calibration_log");
+    });
+  }
+#endif
```

Comment:
As in my previous comment, `dump_config` is triggered when the user starts the API logs, so you don't need to check for any of that.

```suggestion
  if (this->restored_offset_calibration_ || this->restored_power_offset_calibration_ ||
      this->restored_gain_calibration_ || !this->enable_offset_calibration_ || !this->enable_gain_calibration_) {
    this->log_calibration_status_();
  }
```

Reply:
ah, okay, got it